firestore/query.go

```diff
@@ -1238,34 +1234,40 @@ type DocumentIterator struct {
 // always concrete types, and the fact that this one has two different implementations
 // is an internal detail.
 type docIterator interface {
-	next(opts ...RunOption) (*DocumentSnapshot, error)
-	getExplainMetrics() *ExplainMetrics
+	next() (*DocumentSnapshot, error)
+	getExplainMetrics() (*ExplainMetrics, error)
 	stop()
 }
 
+// GetExplainMetrics returns query explain metrics.
+// This is only present when ExplainOptions run option is provided in the query.
```

Comment:
"when [ExplainOptions] is added to the query (see [Query.WithRunOptions]), and after the iterator reaches the end. An error is returned if either of those conditions does not hold."

Reply:
Done